lib/smart_proxy_pulp_plugin/pulp3_api.rb

```diff
@@ -0,0 +1,33 @@
+require 'sinatra'
+require 'smart_proxy_pulp_plugin/pulp3_client'
+require 'smart_proxy_pulp_plugin/disk_usage'
+
+module PulpProxy
+  class Pulp3Api < Sinatra::Base
+    helpers ::Proxy::Helpers
+
+    get "/status" do
+      content_type :json
+      begin
+        result = Pulp3Client.get("api/v3/status/")
+        return result.body if result.is_a?(Net::HTTPSuccess)
+        log_halt result.code, "Pulp server at #{::PulpProxy::Settings.settings.pulp_url} returned an error: '#{result.message}'"
+      rescue Errno::ECONNREFUSED => e
+        log_halt 503, "Pulp server at #{::PulpProxy::Settings.settings.pulp_url} is not responding"
+      rescue SocketError => e
+        log_halt 503, "Pulp server '#{URI.parse(::PulpProxy::Settings.settings.pulp_url.to_s).host}' is unknown"
```

Comment:
Isn't a timeout also a socket error? In that case the message is incorrect. Perhaps we should log the exact error so you can check the logs if there's weirdness going on?

Reply:
a lot of this was copied from the pulp2 api, but I think this improvement makes sense